### DISCUSSION TITLE
Trigger field validation of on property through all field array items

### DISCUSSION THREAD

**Discussion by JohannesKlauss** [react-hook-form/react-hook-form#5576] [Q&A]
I have a field array like this:

```ts
const { append, remove, fields } = useFieldArray<CampaignFormValues, 'assortmentNeeds'>({
  name: 'assortmentNeeds',
})
```

And `assortmentNeeds` has the following values:

```ts
export interface AssortmentNeeds {
  campaignId: string
  productGroup?: string
  category?: string
  demand?: number
  demandRatio?: number
}
```

Now when one `demandRatio` value changes, I want to trigger the validation for all `demandRatio`s of all items of the field array.

So, ideally I'd like to have something like this in a hook:

```ts
const { trigger, control } = useFormContext<CampaignFormValues>()

const assortmentNeeds = useWatch<CampaignFormValues, `assortmentNeeds.${number}`>({
  name: `assortmentNeeds.${index}` as const,
  control,
}) as AssortmentNeeds

useEffect(() => {
  trigger('assortmentNeeds.0...10.demandRatio')
}, [trigger, assortmentNeeds.demandRatio])
```

But of course `0...10` is not a valid index. How can I trigger the `demandRatio` field of all field items?

**Comment by bluebill1049** [comment]
you can upvote this feature request: https://github.com/react-hook-form/react-hook-form/issues/2381

**Reply by JohannesKlauss** [reply]
But isn't this a different one? For one it's closed and also I already can trigger a single field validation of a field array with `trigger('assortmentNeeds.${index}.demandRatio')`. But my idea is that when I don't give an index, the `trigger` function iterates over the length of the array and triggers `demandRatio` for each item in the field array.

**Reply by bluebill1049** [reply]
perhaps you want to do that within a `forEach` then with `trigger`
